src/cockpit/channel.py

```diff
@@ -247,9 +248,22 @@ def create_transport(self, loop: asyncio.AbstractEventLoop, options: Dict[str, o
         raise NotImplementedError
 
     def do_open(self, options):
+        self.freeze_endpoint()
```

Comment:
I have to say I'm not super thrilled about this, for the non-async case...

that being said, I think our eventual plan ought to be that all Transport subclass creation is *async*, so I think it does make sense....